File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.c
int chalkReplacer(int* chalk, int chalkSize, int k) {
    int sum = 0;
    for (int i = 0; i < chalkSize; i++){
        if ((k-sum) < chalk[i]){
            return i;
        }
        sum += chalk[i];
    }
    k = k%sum;
    sum = 0;
    int ret = -1;
    for (int i = 0; i < chalkSize; i++){
        if ((k-sum) < chalk[i]){
            ret = i;
            break;
        }
        sum += chalk[i];
    }
    return ret;
}

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        s: int = 0
        for i in range(len(chalk)):
            if (k-s) < chalk[i]:
                return i
            s += chalk[i]

        k, s, ret = k%s, 0, -1

        for i in range(len(chalk)):
            if (k-s) < chalk[i]:
                ret = i
                break
            s += chalk[i]

        return ret
        